src/DataViews/FunctionsDataViewTest.cpp

```diff
@@ -0,0 +1,655 @@
+
+  is_function_selected = {false, true, false};
+  is_frame_track_enabled = {false, false, false};
+  EXPECT_THAT(view_.GetContextMenu(0, {0, 1, 2}), can_hook_and_unhook());
+  EXPECT_THAT(view_.GetContextMenu(0, {0, 1, 2}), can_enable_frame_tracks());
+
+  is_function_selected = {false, true, false};
+  is_frame_track_enabled = {false, true, false};
+  EXPECT_THAT(view_.GetContextMenu(0, {0, 1, 2}), can_hook_and_unhook());
+  EXPECT_THAT(view_.GetContextMenu(0, {0, 1, 2}), can_enable_and_disable_frame_tracks());
+
+  is_function_selected = {false, true, true};
+  is_frame_track_enabled = {false, true, false};
+  EXPECT_THAT(view_.GetContextMenu(0, {0, 1, 2}), can_hook_and_unhook());
+  EXPECT_THAT(view_.GetContextMenu(0, {0, 1, 2}), can_enable_and_disable_frame_tracks());
+}
+
+TEST_F(FunctionsDataViewTest, GenericDataExportFunctionShowCorrectData) {
+  // This functionality is not tested in this test case.
+  EXPECT_CALL(app_, IsFunctionSelected)
+      .Times(testing::AnyNumber())
+      .WillRepeatedly(testing::Return(false));
+
+  // This functionality is not tested in this test case.
+  EXPECT_CALL(app_, IsFrameTrackEnabled)
+      .Times(testing::AnyNumber())
+      .WillRepeatedly(testing::Return(false));
+
+  // This functionality is not tested in this test case.
+  EXPECT_CALL(app_, HasCaptureData)
+      .Times(testing::AnyNumber())
+      .WillRepeatedly(testing::Return(false));
+
+  view_.AddFunctions({&functions_[0]});
+
+  std::vector<std::string> context_menu = view_.GetContextMenu(0, {0});
+
+  // Copy Selection
+  {
+    const auto copy_selection_idx =
+        std::find(context_menu.begin(), context_menu.end(), "Copy Selection") -
+        context_menu.begin();
+    ASSERT_LT(copy_selection_idx, context_menu.size());
+
+    std::string clipboard;
+    EXPECT_CALL(app_, SetClipboard).Times(1).WillOnce(testing::SaveArg<0>(&clipboard));
+    view_.OnContextMenu("Copy Selection", static_cast<int>(copy_selection_idx), {0});
+    EXPECT_EQ(clipboard, absl::StrFormat(
+                             "Hooked, Function, Size, Module, Address in module\n"
+                             ", %s, %d, %s, %#x\n",
+                             functions_[0].pretty_name(), functions_[0].size(),
+                             std::filesystem::path{functions_[0].module_path()}.filename().string(),
+                             functions_[0].address()));
+  }
+
+  // Export to CSV
+  {
+    const auto export_to_csv_idx =
+        std::find(context_menu.begin(), context_menu.end(), "Copy Selection") -
+        context_menu.begin();
+    ASSERT_LT(export_to_csv_idx, context_menu.size());
+
+    ErrorMessageOr<orbit_base::TemporaryFile> temporary_file_or_error =
+        orbit_base::TemporaryFile::Create();
+    ASSERT_THAT(temporary_file_or_error, orbit_base::HasNoError());
+    const std::filesystem::path temporary_file_path = temporary_file_or_error.value().file_path();
+    temporary_file_or_error.value().CloseAndRemove();
```

Comment:
I was confused, why this file is closed and removed before the context menu action can write to it. I think I understood now, that you actually only need a tmp file path and the detour via creating a temporary file gives you a file path in "/tmp" (or windows equivalent). 

I think as the scope of this PR, there should be a comment explaining that. (The even better solution would be having a tested function in OrbitBase that creates a tmp file path. And this function is then used by TemporaryFile)

Also. Consider deleting the file after this test finishes.

Reply:
`TemporaryFile::CloseAndRemove` exactly does what you suggest. It makes `TemporaryFile` only create a temporary file path and it will still delete the file at the end of the test. We also have multiple similar usages in tests in the codebase. This was surprising to me as well, maybe the API can be improved here but that feels out of scope for this PR.

I will add a comment though. That's a good suggestion.

Reply:
Done.